usageutils/check.go

```diff
@@ -0,0 +1,35 @@
+package usageutils
+
+import (
+	check "github.com/solo-io/go-checkpoint"
+	"math/rand"
+	"time"
+)
+
+// A simple interface for interacting with the checkpoint server, for reporting and version checking
+type UsageClient interface {
+	Start(name, version string)
+}
+
+var _ UsageClient = NewUsageClient()
+
+func NewUsageClient() *usageClient {
+	return &usageClient{}
+}
+
+type usageClient struct {
+}
+
+func (c *usageClient) Start(name, version string) {
+	now := time.Now()
+	check.CallReport(name, version, now)
+	check.CallCheck(name, version, now)
+
+	jitter := time.Duration(rand.Intn(120)) * time.Minute
```

Comment:
ideally jitter should be different in each iteration. maybe something like
```
for {
check.CallCheck
time.Sleep(time.Hour+time.Duration(rand.Intn(120)) * time.Minute)
}
```

see example here: https://github.com/hashicorp/consul/blob/53b25fbe383a42fa1cc80c9883bf0adaa9d79bb6/command/agent/agent.go#L120